astroquery/mast/cutouts.py

```diff
@@ -60,31 +72,50 @@ def _parse_cutout_size(size):
     if np.isscalar(size):
         size = np.repeat(size, 2)
 
+        limit_reached = size[0] > 30 or size[1] > 30
+
     if isinstance(size, u.Quantity):
         size = np.atleast_1d(size)
+
         if len(size) == 1:
             size = np.repeat(size, 2)
 
+        # Based on the literature, TESS resolution is approx. 21 arcseconds per pixel.
+        # We will convert the recommended upper limit for a dimension from pixels
+        # to the unit being passed.
+        unit = size[0].unit
+        upper_limit = (30 * 21*u.arcsec).to(unit).value
```

Comment:
now that there is the `mission` kwarg, too, maybe check for it here, so you only use the TESS hard-wired 21 arcsec pixel scale for TESS?
Also, I think this would not work for pixel units, even though the docstring above says pixel or angular units are both acceptable. ==> overall it would be worth to do parametrized testing using all three options.

Reply:
The `limit_reached` arg will be ignored if `mission` is anything but TESS, and the `upper_limit` check for pixel units is a few lines further up in this method.